pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -278,17 +289,49 @@ func (c *DeploymentConfigController) reconcileDeployments(existingDeployments *k
 }
 
 func (c *DeploymentConfigController) updateStatus(config *deployapi.DeploymentConfig) error {
-	if config.Generation > config.Status.ObservedGeneration {
-		config.Status.ObservedGeneration = config.Generation
+	// NOTE: We should update the status of the deployment config only if we need to, otherwise
+	// we hotloop between updates.
+	if !needsUpdate(config) {
+		return nil
 	}
-	if _, err := c.osClient.DeploymentConfigs(config.Namespace).UpdateStatus(config); err != nil {
+	config.Status.ObservedGeneration = config.Generation
+	if _, err := c.dn.DeploymentConfigs(config.Namespace).UpdateStatus(config); err != nil {
 		glog.V(2).Infof("Cannot update the status for %q: %v", deployutil.LabelForDeploymentConfig(config), err)
-		return transientError(fmt.Sprintf("cannot update the status for %q - requeuing", deployutil.LabelForDeploymentConfig(config)))
+		return wrapError(err)
 	}
 	glog.V(4).Infof("Updated the status for %q (observed generation: %d)", deployutil.LabelForDeploymentConfig(config), config.Status.ObservedGeneration)
 	return nil
 }
 
+func (c *DeploymentConfigController) handleErr(err error, key interface{}) {
+	if err == nil {
```

Comment:
you can have non-nil errors that aren't fatal or transient, you should decide how they're treated.  I'd probably default transient.